### PR TITLE
Implementation for editing an existing regulation.

### PR DESCRIPTION
**Pull request by paulpepper-trade**

### CODE REVIEW

File: regulations/views.py [hunk 1]
```diff
@@ -72,3 +78,46 @@ class RegulationConfirmCreate(TrackedModelDetailView):
     def get_queryset(self):
         tx = WorkBasket.get_current_transaction(self.request)
         return Regulation.objects.approved_up_to_transaction(tx)
+
+
+# TODO:
+# * Should Regulation views apply 'permission_classes' attr.
```
Comment (simonwo) [inline regulations/views.py:84]: Yes :)

File: regulations/views.py [hunk 2]
```diff
@@ -72,3 +78,46 @@ class RegulationConfirmCreate(TrackedModelDetailView):
...
+# * RegulationUpdate inherits BusinessRulesMixin as with other edit views. Which
+#   rules should be applied, if any?
```
Comment (simonwo) [inline regulations/views.py:86]: Oh, good question. Based on reading https://uktrade.github.io/tamato/docs/business_rules/regulations.html, I think:

- ROIMB1
- ROIMB4
- ROIMB8
- ROIMB44
- ROIMB47

File: regulations/views.py [hunk 3]
```diff
@@ -72,3 +78,46 @@ class RegulationConfirmCreate(TrackedModelDetailView):
...
+# * Should business rules be applied to RegulationCreate too?
```
Comment (simonwo) [inline regulations/views.py:87]: Yes, you can probably skip ROIMB8 but all the others are relevant. No harm running ROIMB8 if it's easier to make them the same.

File: regulations/views.py (hunk 1 above)
Comment (paulpepper-trade) [reply regulations/views.py:84]: Ah, actually, my bad. The `permission_classes` is used on DRF's ViewSets not in Django views. Our create and edit views indirectly inherit from `PermissionrequiredMixin` and their base classes define `permission_required = "common.add_trackedmodel"`.

**Comment by simonwo** [comment]
Do we now have a good way to implement some tests for this, a la [here](https://github.com/uktrade/tamato/blob/master/footnotes/tests/test_views.py#L20)?

**Comment by paulpepper-trade** [comment]
> Do we now have a good way to implement some tests for this, a la [here](https://github.com/uktrade/tamato/blob/master/footnotes/tests/test_views.py#L20)?

Not yet...

File: common/tests/factories.py [hunk 4]
```diff
@@ -217,6 +217,17 @@ class BaseRegulationFactory(RegulationFactory):
...
 
+class UIRegulationFactory(BaseRegulationFactory):
+    """
+    Regulation factory used by our UI form tests.
+
+    These are distinct from our other Regulation factories because the required
+    status differs between Regulation model fields and form fields.
+    """
+
+    published_at = Dates().datetime_now
```
Comment (simonwo) [inline common/tests/factories.py:228]: We should prefer use of `date_ranges("now")` if possible.

**Review by simonwo — APPROVED** [review]

File: common/tests/factories.py (hunk 4 above)
Comment (paulpepper-trade) [reply common/tests/factories.py]: Done.

**Comment by paulpepper-trade** [comment]
> > Do we now have a good way to implement some tests for this, a la [here](https://github.com/uktrade/tamato/blob/master/footnotes/tests/test_views.py#L20)?
> 
> Not yet...

Done.

**Comment by paulpepper-trade** [comment]
> > Do we now have a good way to implement some tests for this, a la [here](https://github.com/uktrade/tamato/blob/master/footnotes/tests/test_views.py#L20)?
> 
> Not yet...

Done.

**Pull request merged**